runtime/gc_base/StringTable.cpp

```diff
@@ -554,7 +554,19 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
 	j9object_t charArray;
 	UDATA allocateFlags = J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE;
 	UDATA unicodeLength;
-	bool isCompressable = false;
+
+	bool isCompressible = true;
```

Comment:
This section of code should only set `isCompressible = true` if `IS_STRING_COMPRESSION_ENABLED_VM(vm)` is also true.

Then the repeated checks for `IS_STRING_COMPRESSION_ENABLED_VM(vm)` below can be removed

Reply:
`isCompressible` and `IS_STRING_COMPRESSION_ENABLED_VM(vm)` are not equivalent. The former determines whether the input `U_8* data` is an ASCII string, and `IS_STRING_COMPRESSION_ENABLED_VM(vm)` determines whether `-XX:+CompactStrings` is enabled.

We must decouple these two notions because we want to accelerate the API irregardless of whether compact strings are enabled or not. i.e. if `isCompressible == true` and `IS_STRING_COMPRESSION_ENABLED_VM(vm) == false` we can still accelerate the algorithms for translating from UTF8 to UTF16, except that we will be storing 2-byte characters into the string backing array rather than 1-byte characters.

Reply:
That needs to be described in the code as it's a subtly I had overlooked.  Can you add a descriptive comment to that affect here?

Reply:
This has been fixed with a series of commits starting at 2cb4393 and ending at fc7f52f. Each commit follows a logical step towards the final goal and the commit messages explain the direction. The `isCompressible` notion is now completely removed and replaced by a string flag `J9_STR_ASCII` which is what we really care about.

The string flag represents that a UTF8 string is an ASCII string and we can take advantage of this irregardless of whether compression is enabled or not, as we can see in the various helpers in `VM_VMHelpers`. The code is no much improved in terms of cleanliness, documentation, and implementation.